src/AppInstallerTestExeInstaller/main.cpp

```diff
@@ -1,16 +1,116 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+#include <windows.h>
+#include <winreg.h>
+#include <winerror.h>
 #include <iostream>
 #include <fstream>
 #include <filesystem>
 #include <sstream>
 
-// The installer simply prints all args to an output file
+using namespace std::filesystem;
+
+std::wstring_view registrySubkey = L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\";
+std::wstring_view defaultProductID = L"{A499DD5E-8DC5-4AD2-911A-BCD0263295E9}";
+
+std::wstring GenerateUninstaller() {
+    path tempPath = temp_directory_path();
+    path uninstallerPath = tempPath;
+    uninstallerPath /= "UninstallTestExe.bat";
+
+    std::cout << "Uninstaller located at path: " << uninstallerPath << '\n';
+
+    path uninstallerOutputTextFilePath = tempPath;
+    uninstallerOutputTextFilePath /= "TestExeUninstalled.txt";
+
+    std::ofstream uninstallerScript(uninstallerPath);
+    uninstallerScript << "@echo off\n";
+    uninstallerScript << "ECHO. >" << uninstallerOutputTextFilePath << "\n";
+    uninstallerScript << "ECHO AppInstallerTestExeInstaller.exe uninstalled successfully.\n";
+    uninstallerScript.close();
+
+    return uninstallerPath.wstring();
+}
+
+void WriteToUninstallRegistry(std::wstring productID, std::wstring uninstallerPath) {
+    HKEY hkey;
+    LONG lReg;
+
+    // String inputs to registry must be of wide char type
+    const wchar_t* displayName = L"AppInstallerTestExeInstaller\0";
+    const wchar_t* publisher = L"Microsoft Corporation\0";
+    const wchar_t* uninstallString = uninstallerPath.c_str();
+    DWORD version = 1;
+
+    std::wstring registryKey = (std::wstring)registrySubkey;
+
+    if (!productID.empty()) 
+    {
+        registryKey += productID;
+        std::wcout << "Product Code Overrided to: " << registryKey.c_str() << "\n";
+    }
+    else 
+    {
+        registryKey += defaultProductID;
+        std::wcout << "Default Product Code Used: " << registryKey.c_str() << "\n";
+    }
+
+    lReg = RegCreateKeyEx(
+        HKEY_LOCAL_MACHINE,
+        registryKey.c_str(),
+        0,
+        NULL,
+        REG_OPTION_NON_VOLATILE,
+        KEY_ALL_ACCESS | KEY_WOW64_64KEY,
+        NULL,
+        &hkey,
+        NULL);
+
+    if (lReg == ERROR_SUCCESS) {
+
+        std::cout << "Successfully opened registry key \n";
+
+        // Set Display Name Property Value
+        if (RegSetValueEx(hkey, L"DisplayName", NULL, REG_SZ, (LPBYTE)displayName, (DWORD)(wcslen(displayName) + 1) * sizeof(wchar_t)) != ERROR_SUCCESS)
+        {
+            std::cout << "Failed to write DisplayName value \n";

```

Comment:
>std::cout << "Failed to write DisplayName value \n"; [](start = 12, length = 52)

Typically for debugging purposes it is helpful to also print the return value.

Reply:
Changed to print return value if failure occurs